### DISCUSSION TITLE
Clone through API

### DISCUSSION THREAD

**Discussion by alexbekk** [DependencyTrack/dependency-track#2804] [Q&A] [answered]
I am trying to clone a project with a command:

curl -X PUT "http://IP:8080/api/v1/project/clone" -H  "accept: application/json" -H  "X-Api-Key: 9gf1rnzKHWL9XUwoUaD5MwKLJJHJH" -H  "Content-Type: application/json" -d "{\"project\":\"9caca92b-3f0a-49ed-be82-e19e3570d1c2\",\"version\":\"develop3\"}"

It looks like the project is cloned without the additional things.
![image](https://github.com/DependencyTrack/dependency-track/assets/37347275/8adfb471-bdd0-4b80-a504-e844bd30826d)
How can I include all these artifacts in a cloned project through the API?

**Comment by nscuro** [comment, marked as answer]
Yes, the API has corresponding fields for all of the toggles shown in the UI. 

You can see the full structure of the request body here: https://github.com/DependencyTrack/client-go/blob/v0.9.0/project.go#L144

Look at the `json` part for the actual API field names.

**Reply by nscuro** [reply]
The fields are `boolean` and as such all default to `false` if not provided explicitly.

Another way to see the request the frontend is performing is to use your browser's developer tool. The *Network* tab will show the request.

**Reply by alexbekk** [reply]
Thank you very much!

**Comment by alexbekk** [comment]
curl -X PUT "http://IP:8080/api/v1/project/clone" -H  "accept: application/json" -H  "X-Api-Key: 9gf1rnzKHWL9XUwoUaD5Mwy9ksdBHN" -H  "Content-Type: application/json" -d "{\"project\":\"9caca92b-3f0a-49ed-be82-e19e3570d1c2\",\"version\":\"develop29\",\"includeAuditHistory\":\"true\",\"includeComponents\":\"true\",\"includeProperties\":\"true\",\"includeServices\":\"true\",\"includeTags\":\"true\"}"
